### ISSUE TITLE
Automated tests migration

### ISSUE THREAD

**Issue by diegoliberman** [instedd/cdx#1426]
What we have today:
- The integration tests are located in the `specs/features` directory.
--> They use capybara
--> Poltergeist is a gem that runs capybara tests in a headless browser
--> They use a very old browser PanthomJS

- When upgrading rails, there might be some tests that start failing because the browser we use will require to support very old versions (IE < 11).
--> We can fix the issues in those old browsers.
--> We can migrate the integration tests to a modern browser, before continuing the upgrade.
--> We can disable the failing tests until we finish the upgrade and then we migrate the integration tests.

- Migrating integration tests
--> It would require migrating Poltergeist to a driver based on Selenium/WebDriver
--> An option is Cuprite but only run tests on Google Chrome.

Next steps?
1. Define a driver for migrating the integration tests and replacing Poltergeist.
2. Migrate the driver and make sure tests are working as expected.
3. Resume rails migration.

**Comment by straight-shoota** [comment]
I'm getting a JavaScript error in poltergeist while writing integration specs for #1389. It says `SyntaxError: Parse error` but no location or other info.

In the browser, everything works. So I suspect this is probably due to outdated JavaScipt engine.

**Comment by ysbaddaden** [comment]
@straight-shoota Yes, PhantomJS is very outdated Webkit. You maybe introduced an ES5 or ES6 keyword or construct like `let` or `const`.

**Comment by ysbaddaden** [comment]
I tried to upgrade to Selenium::WebDriver. It has been quite a pain, but I finally got Capybara to talk to a Firefox instance (but couldn't get it to communicate with Chrome). Now, the driver is much more restricted and strict than the PhantomJS driver.

Examples:
- We can't access the HTTP status code for the loaded page (which is normal), so some assertions must be rewritten to verify the page contents.
- We can't select an invisible DOM element, this prevents us from attaching a file to upload (e.g. device manifest) because we hide the input by default.

**Comment by ysbaddaden** [comment]
We can just pass a `:visible` to finders: https://stackoverflow.com/a/43919728/199791 and it works :tada:

**Comment by ysbaddaden** [comment]
While investigating why some Cucumber scenarios failed in `alert.feature`, I noticed that some select options are currently _never_ selected because of a PhantomJS bug. I re-enabled the selection, and some specs broke because the user to receive alerts isn't listed in the cdx_select options (oops). I disabled those scenarios.

I was able to fix the issues above. I added a few `:visible => :all` options (not much), and modified `success?` to verify that the user menu is available on the page instead of checking the HTTP status code.

I'm down to one failure.
